scripts/release.py

```diff
@@ -0,0 +1,256 @@
+        return "master"
+
+
+def create_release_branch(version: Text) -> Text:
+    """Create a new branch for this release. Returns the branch name."""
+
+    branch = f"{RELEASE_BRANCH_PREFIX}{version}"
+    check_call(["git", "checkout", "-b", branch])
+    return branch
+
+
+def create_commit(version: Text) -> None:
+    """Creates a git commit with all stashed changes."""
+    check_call(["git", "commit", "-m", f"prepared release of version {version}"])
+
+
+def push_changes() -> None:
+    """Pushes the current branch to origin."""
+    check_call(["git", "push", "origin", "HEAD"])
+
+
+def ensure_clean_git() -> None:
+    """Makes sure the current working git copy is clean."""
+
+    try:
+        check_call(["git", "diff-index", "--quiet", "HEAD", "--"])
+    except CalledProcessError:
+        print("Your git is not clean. Release script can only be run from a clean git.")
+        sys.exit(1)
+
+
+def parse_next_version(version: Text) -> Text:
+    """Find the next version as a proper semantic version string."""
+    if version == "major":
+        return str(Version.coerce(get_current_version()).next_major())
+    elif version == "minor":
+        return str(Version.coerce(get_current_version()).next_minor())
+    elif version == "patch":
+        return str(Version.coerce(get_current_version()).next_patch())
+    elif semantic_version.validate(version):
+        return version
+    else:
+        raise Exception(f"Invalid version number '{cmdline_args.next_version}'.")
+
+
+def next_version(args: argparse.Namespace) -> Text:
+    """Take cmdline args or ask the user for the next version and return semver."""
+    return parse_next_version(args.next_version or ask_version())
+
+
+def generate_changelog(version: Text) -> None:
+    """Call towncrier and create a changelog from all available changelog entries."""
+    check_call(["towncrier", "--yes", "--version", version], cwd=str(project_root()))
+
+
+def print_done_message(branch: Text, base: Text, version: Text) -> None:
+    """Print final information for the user on what to do next."""
+
+    pull_request_url = f"{REPO_BASE_URL}/compare/{base}...{branch}?expand=1"
+
+    print()
+    print(f"\033[94m All done - changes for version {version} are ready! \033[0m")
+    print()
+    print(f"Please open a PR on GitHub: {pull_request_url}")
+
+
+def main(args: argparse.Namespace) -> None:
+    """Start a release preparation."""
+
+    print(
+        "The release script will increase the version number, "
+        "create a changelog and create a release branch. Let's go!"
+    )
+
+    ensure_clean_git()
+    version = next_version(args)
+    confirm_version(version)
+
+    write_version_file(version)
+    write_version_to_pyproject(version)
+
+    generate_changelog(version)
+    base = git_current_branch()
+    branch = create_release_branch(version)
+
+    create_commit(version)
+    push_changes()
+
+    print_done_message(branch, base, version)
+
+
+if __name__ == "__main__":
```

Comment:
could you please quickly point me to the changed parts of the script in comparison to rasa / rasa x , then I can focus the review on these parts

Reply:
the only thing I added here is `write_version_to_pyproject`
everything else is done by @tmbo in this PR: https://github.com/RasaHQ/rasa-sdk/pull/146